tuf/tuf.go

```diff
@@ -173,6 +173,132 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetBaseRole gets a base role from this repo's metadata
+func (tr *Repo) GetBaseRole(name string) (data.BaseRole, error) {
+	if !data.IsBaseRole(name) {
+		return data.BaseRole{}, fmt.Errorf("Invalid base role name specified: %s", name)
```

Comment:
Non-blocking - I'm not sure if this applies here, but `NotaryRepository` uses `data.ErrInvalidRole` whenever a role is supplied with an invalid name - not sure if that'd be a useful error type to return here and in `GetDelegationRole`